src/pages/index.js

```diff
@@ -21,103 +23,251 @@ import {
+}
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, cardData]) => {
+    userInfo.setUserInfo({
+      title: userData.name,
+      description: userData.about,
+    });
+    userInfo.setAvatarInfo(userData.avatar);
+    userId = userData._id;
+    cardSection = new Section(
+      {
+        data: cardData,
+        renderer: renderCard,
+      },
+      cardListSelector
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
 export const cardTemplate = document
   .querySelector("#card-template")
   .content.querySelector(".card");
 
+/********************
+ * POPUP WITH IMAGE *
+ ********************/
+
 const modalWithImage = new PopupWithImage({
-  popupSelector: imageModalSelector,
+  modalSelector: imageModalSelector,
+});
+
+const avatarModal = new PopupWithForm({
+  modalSelector: modalChangeProfileAvatarSelector,
+  handleFormSubmit: (data) => {
+    avatarModal.renderLoading(true);
+    api
+      .updateUserProfile({ avatar: data.url })
+      .then((data) => {
+        userInfo.setAvatarInfo(data.avatar);
+        avatarModal.close();
+      })
+      .catch(console.error)
+
+      .finally(() => {
+        avatarModal.renderLoading(false);
+      });
+  },
+  loadingText: "Saving...",
 });
 
-const modalWithFormUser = new PopupWithForm({
-  popupSelector: profileModalSelector,
-  handleFormSubmit: (cardData) => {
-    userInfo.setUserInfo(cardData);
+// console.log('profileModalSelector: ', profileModalSelector)
+
+const modalFormUser = new PopupWithForm({
+  modalSelector: profileModalSelector,
+  handleFormSubmit: (data) => {
+    modalFormUser.renderLoading(true);
+    api
+      .updateUserInfo(data)
+      .then((data) => {
+        userInfo.setUserInfo({
+          title: data.name,
+          description: data.about,
+        });
+
+        modalFormUser.close();
+      })
+      .catch(console.error)
+
+      .finally(() => {
+        modalFormUser.renderLoading(false);
+      });
   },
+  loadingText: "Saving...",
 });
 
 const modalFormImage = new PopupWithForm({
-  popupSelector: cardModalSelector,
+  modalSelector: cardModalSelector,
   handleFormSubmit: (inputValues) => {
-    const name = inputValues.title;
-    const link = inputValues.url;
+    modalFormImage.renderLoading(true);
+    // console.log('inputValues: ', inputValues)
 
-    renderCard({ link, name }, cardList);
+    api
+      .addCard(inputValues)
+      .then((inputValues) => {
+        renderCard(inputValues);
+        modalFormImage.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        modalFormImage.renderLoading(false);
+      });
   },
+
+  loadingText: "Saving...",
 });
 
-const cardSection = new Section(
-  {
-    data: initialCards,
-    renderer: renderCard,
+const deleteModal = new PopupWithConfirmation({
+  handleFormSubmit: () => {
+    deleteModal._handleFormSubmit();
```

Comment:
Please remove this, you should not use private methods outside of a class and `PopupWithConfirmation` should not have `handleFormSubmit` argument, it has `setSubmitAction` for updating the handler

```js
const deleteModal = new PopupWithConfirmation({
  modalSelector: cardModalDelete,
  loadingText: "Deleting...",
})
```